var/spack/repos/builtin/packages/amdscalapack/package.py

```diff
@@ -37,7 +38,9 @@ class Amdscalapack(ScalapackBase):
               msg="ILP64 is supported from 3.1 onwards")
 
     def url_for_version(self, version):
-        if version == Version('3.1'):
+        if version == Version('3.2'):
+            return "https://github.com/amd/aocl-scalapack/archive/3.2.tar.gz"
+        elif version == Version('3.1'):
```

Comment:
I agree somewhat with #31521 about the version handling here needing to be changed.  For example, this package could use something along the following for the body of this method:

```
        vers = "https://github.com/amd/{0}/archive/{1}.tar.gz"
        if version >= Version('3.1'):
            return vers.format('aocl-scalapack', version)

        return vers.format('scalapack', version)
```

Reply:
Changes incorporated